### PR TITLE
Plastic flaps are now airtight

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
tweak: Plastic flaps are now airtight.
/:cl:

Right now, some maps (for example Delta) atmos connect huge areas through plastic flaps. 
You break a window in room A and can basically hear the CPU screech as it updates room A, corridors B, C, and D as well as the connected rooms.

An alternative solution would be adding windoors in front of plastic flaps, like how it's done on Boxstation.

What do you prefer?

### CODE REVIEW

**Comment by Supermichael777** [comment]
plastic flaps were a hand-wave for mule-bots.  I see mulebots used a lot less these days.  Their outdated with disposal shipping.  their vulnerable to interception, slow, and limited in scale.

**Comment by Iatots** [comment]
Nobody uses disposal for delivery either.

**Comment by Rinoka** [comment]
People use disposals more than mulebots; disposals are faster, simpler, and don't require pulling up a clunky interface with low control when you can generally trust disposals to function for your needs, especially since a failed delivery for whatever reason will just end up at your door again.

As it is, flaps are nice, but they really don't need to allow airflow.

**Comment by chesse20** [comment]
ad this please :)

File: code/game/objects/structures/plasticflaps.dm [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+	name = "airtight plastic flaps"
+	desc = "Heavy duty, airtight, plastic flaps. Definitely can't get past those. No way."
 	icon = 'icons/obj/stationobjs.dmi'
 	icon_state = "plasticflaps"
 	armor = list("melee" = 100, "bullet" = 80, "laser" = 80, "energy" = 100, "bomb" = 50, "bio" = 100, "rad" = 100, "fire" = 50, "acid" = 50)
 	density = FALSE
 	anchored = TRUE
 	layer = ABOVE_MOB_LAYER
 	var/state = PLASTIC_FLAPS_NORMAL
+	CanAtmosPass = ATMOS_PASS_NO
```
Comment (Cyberboss) [inline code/game/objects/structures/plasticflaps.dm]: above `state` please

File: code/game/objects/structures/plasticflaps.dm [hunk 2]
```diff
@@ -92,17 +93,13 @@
...
 
-/obj/structure/plasticflaps/mining //A specific type for mining that doesn't allow airflow because of them damn crates
-	name = "airtight plastic flaps"
-	desc = "Heavy duty, airtight, plastic flaps."
-	CanAtmosPass = ATMOS_PASS_NO
-
-/obj/structure/plasticflaps/mining/New()
+/obj/structure/plasticflaps/New()
 	air_update_turf(1)
 	. = ..()
```
Comment (Cyberboss) [inline code/game/objects/structures/plasticflaps.dm]:
```dm
/obj/structure/plasticflaps/Initialize()
 	. = ..()
 	air_update_turf(TRUE)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by nicbn** [comment]
Windoors please

**Comment by Cyberboss** [comment]
Pretty sure windoors are already air tight

**Review by Cyberboss — APPROVED** [review]
To whomever merges this, add flaps to the "Resprite Needed" project

**Comment by nicbn** [comment]
I'm asking him to put windoors in the maps instead of making the flaps airtight.

**Comment by Cyberboss** [comment]
@MapDiffBot

**Comment by ExcessiveUseOfCobblestone** [comment]
There's balance implications here but idt they should be an issue.

**Comment by Rinoka** [comment]
really the only balance change i can think of is that this might make a personal glitch I kind of enjoy (using air canisters to slide under flaps) just trap you instead

but that requires several things to make happen so

**Pull request merged**